### PR TITLE
#4508 writing tests for brokerEpics

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4508)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/api/__mocks__/brokerApi.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+// simulate a promise request with delay 10ms
+export const stop = (): Observable<BasicResponse> =>
+  of({
+    status: 200,
+    title: 'mock stop broker data',
+    data: { ...omit(runtimeData, 'state') },
+  }).pipe(
+    // to simulate broker is "stopped" in fetch request
+    tap(() => !!runtimeData.state && delete runtimeData.state),
```
Comment (konekoya) [inline ohara-manager/client/src/api/__mocks__/brokerApi.ts:111]: I think you don't need to do a check before deleting?

Comment (saivirtue) [reply ohara-manager/client/src/api/__mocks__/brokerApi.ts:111]: Just double-check 😄 
I will remove this line, thanks for suggestion

File: ohara-manager/client/src/store/epics/broker/stopBrokerEpic.js [hunk 2]
```diff
@@ -17,37 +17,47 @@
...
-        return res.data;
+        else return res.data;
       }),
       map(data => normalize(data, schema.broker)),
       map(normalizedData => merge(normalizedData, { brokerId })),
       map(normalizedData => actions.stopBroker.success(normalizedData)),
-      retryWhen(error => error.pipe(delay(2000), take(10))),
+      retryWhen(error =>
+        error.pipe(
+          concatMap((e, i) =>
```
Comment (konekoya) [inline ohara-manager/client/src/store/epics/broker/stopBrokerEpic.js:52]: I think we should probably go with a naming convention with the error object... we're now using `error`, `err`, `e` throughout the epic code @eechih @wu87988622

File: ohara-manager/client/src/store/epics/broker/stopBrokerEpic.js [hunk 3]
```diff
@@ -62,6 +72,6 @@ export default action$ =>
   action$.pipe(
     ofType(actions.stopBroker.TRIGGER),
     map(action => action.payload),
-    debounce(() => interval(1000)),
-    switchMap(params => stopBroker$(params)),
+    distinctUntilChanged(),
```
Comment (konekoya) [inline ohara-manager/client/src/store/epics/broker/stopBrokerEpic.js:75]: 👍

File: ohara-manager/client/src/store/epics/broker/stopBrokerEpic.js (hunk 2 above)
Comment (saivirtue) [reply ohara-manager/client/src/store/epics/broker/stopBrokerEpic.js:52]: How about following the js doc?
`retryWhen(errors => ...)`
`errors.pipe(concatMap(value, index)...))`

Comment (eechih) [reply ohara-manager/client/src/store/epics/broker/stopBrokerEpic.js:52]: > How about following the js doc?
retryWhen(errors => ...)
errors.pipe(concatMap(value, index)...))

+1

Comment (konekoya) [reply ohara-manager/client/src/store/epics/broker/stopBrokerEpic.js:52]: Same for me :)

File: ohara-manager/client/src/store/epics/__tests__/createBrokerEpic.test.js [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+import { TestScheduler } from 'rxjs/testing';
+
+import createBrokerEpic from '../broker/createBrokerEpic';
+import { entity as brokerEntity } from 'api/__mocks__/brokerApi';
+import * as actions from 'store/actions';
+import { getId } from 'utils/object';
+
+jest.mock('api/brokerApi');
+
+const testScheduler = () =>
```
Comment (eechih) [inline ohara-manager/client/src/store/epics/__tests__/createBrokerEpic.test.js:26]: Can we rename to `makeTestScheduler`?

Comment (saivirtue) [reply ohara-manager/client/src/store/epics/__tests__/createBrokerEpic.test.js:26]: sure~

**Pull request merged**